js/reliable_trigger.js

```diff
@@ -0,0 +1,115 @@
+    if (!name) return true;
+
+    // Handle event maps.
+    if (typeof name === 'object') {
+      for (var key in name) {
+        obj[action].apply(obj, [key, name[key]].concat(rest));
+      }
+      return false;
+    }
+
+    // Handle space separated event names.
+    if (eventSplitter.test(name)) {
+      var names = name.split(eventSplitter);
+      for (var i = 0, l = names.length; i < l; i++) {
+        obj[action].apply(obj, [names[i]].concat(rest));
+      }
+      return false;
+    }
+
+    return true;
+  };
+
+  // A difficult-to-believe, but optimized internal dispatch function for
+  // triggering events. Tries to keep the usual cases speedy (most internal
+  // Backbone events have 3 arguments).
+  var triggerEvents = function(events, name, args) {
+    var ev, i = -1, l = events.length, a1 = args[0], a2 = args[1], a3 = args[2];
+    switch (args.length) {
+      case 0:
+        while (++i < l) {
+          try {
+            (ev = events[i]).callback.call(ev.ctx);
+          }
+          catch (error) {
+            console.log('Model caught error triggering', name, 'event:', error && error.stack ? error.stack : error);
+          }
+        }
+        return;
+      case 1:
+        while (++i < l) {
+          try {
+            (ev = events[i]).callback.call(ev.ctx, a1);
+          }
+          catch (error) {
+            console.log('Model caught error triggering', name, 'event:', error && error.stack ? error.stack : error);
+          }
+        }
+        return;
+      case 2:
+        while (++i < l) {
+          try {
+            (ev = events[i]).callback.call(ev.ctx, a1, a2);
+          }
+          catch (error) {
+            console.log('Model caught error triggering', name, 'event:', error && error.stack ? error.stack : error);
+          }
+        }
+        return;
+      case 3:
+        while (++i < l) {
+          try {
+            (ev = events[i]).callback.call(ev.ctx, a1, a2, a3);
+          }
+          catch (error) {
+            console.log('Model caught error triggering', name, 'event:', error && error.stack ? error.stack : error);
+          }
+        }
+        return;
+      default:
+        while (++i < l) {
+          try {
+            (ev = events[i]).callback.apply(ev.ctx, args);
+          }
+          catch (error) {
+            console.log('Model caught error triggering', name, 'event:', error && error.stack ? error.stack : error);
+          }
+        }
+        return;
+    }
+  };
+
+  Backbone.Model.prototype.trigger = Backbone.View.prototype.trigger = Backbone.Events.trigger = function(name) {
+    if (!this._events) return this;
+    var args = slice.call(arguments, 1);
+    if (!eventsApi(this, 'trigger', name, args)) return this;
+    var events = this._events[name];
+    var allEvents = this._events.all;
+    if (events) triggerEvents(events, name, args);
+    if (allEvents) triggerEvents(allEvents, name, arguments);
+    return this;
```

Comment:
I agree that it would reduce the size of the change, but it would also prevent all listeners from hearing about the change. I think it's worth the additional complexity.